File: Blind_75_LeetCode_Questions/Maximum_Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) 
    {
        int n = nums.size();
        int max_num = INT_MIN;
        int result = 0;

        for (int i = 0; i < n; i++)
        {
            result = result + nums[i];
            max_num = max(max_num,result);
            
            if(result < 0)
                result = 0;
        }
        return max_num;
        
    }
};

File: Blind_75_LeetCode_Questions/java/Array/Q53_Maximum_Subarray/Maximum_Subarray.java
class Solution {
    public int maxSubArray(int[] nums) 
    {
        int len = nums.length;
        int result = nums[0];
        int max_num = nums[0];

        for(int i = 1; i < len; ++i)
        {
            max_num = Math.max(max_num + nums[i], nums[i]);
            result = Math.max(max_num, result);

        }
        return result;
        
    }
}
